### DISCUSSION TITLE
E-trans Keyboard

### DISCUSSION THREAD

**Discussion by sofia-bg** [UNC-Project-Open-AAC/OS-DPI#169] [Show and tell]
Based on a conversation today about quadrant arrangement work Krista Wilkinson has done with symbol based layouts, I made two replications of an e-trans board style keyboard (i.e., alphabet broken into quadrants and display in the middle). **Layout tabs are ugly**, but kept adding elements until I got the visual layout I wanted.

<img width="1489" alt="Screen Shot 2023-03-01 at 12 42 56 AM" src="https://user-images.githubusercontent.com/92817124/222055066-ee37c9fc-4448-4fd1-87da-058e2146cc82.png">

[Version with clear/space/speak](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/50ac262e7d26c41164628f31f7e990a3a38ffd6c/Designs%20In%20Progress/etrans/e_tranv2.osdpi)

[Version WITHOUT clear/space/speak](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/50ac262e7d26c41164628f31f7e990a3a38ffd6c/Designs%20In%20Progress/etrans/e_tranv3.osdpi)

**Comment by gbishop** [comment]
This looks great! Boy we sure need highlights of UI components from the
designer interface! It's on my list.

You did a great job on that layout!

**Reply by sofia-bg** [reply]
Well and wondering about being able to move up/down without having to go back to edit menu before each advance up/down? I assume it is that way for accessibility for folks who are using keyboard?

Also, hadn't seen the gap function before last night! I really like it!

**Comment by gbishop** [comment]
We could bind another key. For example, Control(Option)-ArrowUp might be
move up.

Or you can Cut and Paste. So you'd cut with Alt-E C C Enter, move with up
or down arrow and then paste with Alt-E P Enter.

**Reply by sofia-bg** [reply]
can we cut and paste items between stacks? That was where I was doing most of the moving last night, either adding new stacks or new gaps between stacks.

**Comment by gbishop** [comment]
Yes. If it is working correctly you should be able to cut from one stack
and paste into another. You can even paste into a design in a different tab
in your browser! And not just stacks. It should let you paste anywhere that
the component is allowed.

Then if you want to get REALLY crazy you can copy a component. Paste it
into an email and send it to someone else. You can copy it from the email
and paste the result into a design. Of course some stuff could go wrong if
the email program messes with the text in any way.

I hallucinate that we could make a library of preconfigured components that
folks can copy from.
